sreejasuresh007-P1.md

```diff
@@ -0,0 +1,182 @@
+
+**‘port’**: port number of database;
+
+**‘database’**: database name in which data has to be stored;
+
+1. Username is important as there are multiple users who are given different kinds of access to the database. This is done to increase security levels of data. Similarly multiple databases can be used to increase data security and division. 
+
+```
+connection = psycopg2.connect(user = "username", password = "password", host = "hostname", port = "port_number", database = "database")
+```
+
+2. Cursor objects are created to execute operations on the database. Execute method is used for executing commands. **cursor** method creates an object for handling operations on the database.
+
+```
+cur=connection.cursor()
+```
+
+3. Query for creating a table. This table has four columns stu_id, stu_firstname, stu_lastname, and department
+
+```
+create_query = """CREATE TABLE student(stu_id integer PRIMARY KEY, stu_firstname text, stu_lastname text, department text)"""   
+```
+
+4. Query for inserting values into database. 
+
+```
+insert_query = """INSERT INTO student VALUES (%s, %s, %s, %s)", (189076, 'Alice', 'Bob', 'IT')""" 
+cur.execute(insert_query) 
+```
+
+5. To read a csv file use open method. ‘r’ denotes read mode. Reader method creates an object to iterate through the file. Next method is used to skip the header row. This method needs to use csv module to read file contents.
+
+```
+with open(‘student_details.csv', 'r') as filename:
+  fileReader = csv.reader(filename)
+  next(fileReader)
+```
+
+6. For iterating through the file contents and inserting it into database.
+
+```
+for row in fileReader:
+  cur.execute("""INSERT INTO student VALUES (%s, %s, %s, %s)""", row)
+```
+
+7. More optimized insert would be to use copy_from method for loading data without looping over INSERT command. The data is loaded without execute command as well. copy_from directly loads data 
+
+```
+with open('user_accounts.csv', 'r') as f:
+   next(f) 
+   cur.copy_from(f, 'users', sep=',')
+connection.commit()
+```
+
+8. Query for selecting all values of the table. **fetchall** method retrieves all records 
+
+```
+select_query = """select * from STUDENT"""
+cur.execute(select_query)
+res_allrecord = cur.fetchall()
+```
+
+9. Query for selecting values of the table based on stu_id. fetchone method retrieves the record based on given criteria
+
+```
+select_query = """select * from STUDENT where stu_id = %s"""
+cur.execute(select_query, ([stu_id]))
+res_record = cur.fetchone()
+```
+
+10. Query for updating value based on stu_id
+
+```
+update_query = """Update STUDENT set department = %s where stu_id = %s"""    
+cur.execute(update_query, ([department, stu_id] ))
+```
+
+11.Query for deleting value based on stu_id
+
+```
+delete_query = """Delete from STUDENT where stu_id = %s"""
+cur.execute(delete_query, ([stu_id,] ))
+```
+
+12. Close the cursor and connection
+
+```
+cur.close()
+connection.close()
+```
+
+**Complete python code**
+
+```
```

Comment:
Add the language tag to the code block for proper syntax highlighting like [this](https://github.com/blrgis/blrgis.github.io/pull/1/files#diff-ba4b79cd400f294effc0b7536c5c3ee2R13)

Reply:
Done